extensions/amp-apester-media/0.1/utils.js

```diff
@@ -0,0 +1,105 @@
+
+import * as events from '../../../src/event-helper';
+
+const rules = [
+  // if it says it's a webview, let's go with that
+  'WebView',
+  // iOS webview will be the same as safari but missing "Safari"
+  '(iPhone|iPod|iPad)(?!.*Safari)',
+  // Android Lollipop and Above: webview will be the same as native but it will contain "wv"
+  // Android KitKat to lollipop webview will put {version}.0.0.0
+  'Android.*(wv|\.0\.0\.0)',
+  // old chrome android webview agent
+  'Linux; U; Android',
+];
+const webviewRegExp = new RegExp('(' + rules.join('|') + ')', 'ig');
+
+function isWebview(ua) {
+  return !!ua.match(webviewRegExp);
+}
+
+/**
+ * Gets the user platform.
+ * @returns {string}
+ */
+export function getPlatform() {
+  const webview = isWebview(navigator.userAgent);
+  return `mobile${webview ? '-webview' : ''}`.toLowerCase();
+}
+
+/**
+ * Extracts tags from a given element .
+ * @param element
+ */
+function extractElementTags(element) {
+  const tagsAttribute = element.getAttribute('data-apester-tags');
+  if (tagsAttribute) {
+    return tagsAttribute.split(',').map(tag => tag.trim()) || [];
+  }
+  return [];
+}
+
+/**
+ * Extracts meta tags from the document.
+ * @return {Array} array of keywords
+ */
+function extractMetaTags(doc) {
+  const selectors = ['meta[name=keywords]', 'meta[name=news_keywords]',
```

Comment:
/cc @rudygalfi @dvoytenko are extensions allowed to read and pass this data?

Reply:
We are using this data in order to get contextual content for the required page.
The content is picked from the publisher contents pool.

Reply:
Got it. but AMP Cache removes these metadata to optimize the size of the page so in majority of cases you won't find them anyway. 

If this is a critical feature, please create an issue so the team can assess: 1- have cache not remove them 2- decide whether it is fine to allow 3Ps to pass this data to themselves.